rig-neo4j/examples/vector_search_simple.rs

```diff
@@ -0,0 +1,138 @@
+//! Simple end-to-end example of the vector search capabilities of the `rig-neo4j` crate.
+//! This example expects a running Neo4j instance running.
+//! It:
+//! 1. Generates embeddings for a set of 3 "documents"
+//! 2. Adds the documents to the Neo4j DB
+//! 3. Creates a vector index on the embeddings
+//! 4. Queries the vector index
+//! 5. Returns the results
+use std::env;
+
+use rig::{
+    embeddings::EmbeddingsBuilder,
+    providers::openai::{Client, TEXT_EMBEDDING_ADA_002},
+    vector_store::VectorStoreIndex as _,
+};
+use rig_neo4j::{
+    vector_index::{IndexConfig, SearchParams},
+    Neo4jClient, ToBoltType,
+};
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // Initialize OpenAI client
+    let openai_api_key = env::var("OPENAI_API_KEY").expect("OPENAI_API_KEY not set");
+    let openai_client = Client::new(&openai_api_key);
+
+    // Initialize Neo4j client
+    let neo4j_uri = env::var("NEO4J_URI").expect("NEO4J_URI not set");
+    let neo4j_username = env::var("NEO4J_USERNAME").expect("NEO4J_USERNAME not set");
+    let neo4j_password = env::var("NEO4J_PASSWORD").expect("NEO4J_PASSWORD not set");
+
+    let neo4j_client = Neo4jClient::connect(&neo4j_uri, &neo4j_username, &neo4j_password).await?;
+
+    // Select the embedding model and generate our embeddings
+    let model = openai_client.embedding_model(TEXT_EMBEDDING_ADA_002);
+
+    let embeddings = EmbeddingsBuilder::new(model.clone())
+        .simple_document("doc0", "Definition of a *flurbo*: A flurbo is a green alien that lives on cold planets")
+        .simple_document("doc1", "Definition of a *glarb-glarb*: A glarb-glarb is a ancient tool used by the ancestors of the inhabitants of planet Jiro to farm the land.")
+        .simple_document("doc2", "Definition of a *linglingdong*: A term used by inhabitants of the far side of the moon to describe humans.")
+        .build()
+        .await?;
+
+    // The struct that will reprensent a node in the database. Used to deserialize the results of the query (passed to the `top_n` methods)
+    // ❗IMPORTANT: The field names must match the property names in the database
+    #[derive(serde::Deserialize)]
+    struct Document {
+        #[allow(dead_code)]
+        id: String,
+        document: String,
+        #[allow(dead_code)]
+        embedding: Vec<f32>,
+    }
+
+    let _create_nodes = futures::future::join_all(embeddings.into_iter().map(|doc| {
```

Comment:
This is ok, but using `Streams` from the `futures` crate is usually preferred (`Streams` are like the async version of `Iterators`, i.e.: closures can be async)

Reply:
will change it to streams!